### PR TITLE
Thirdparty api tests

### PR DESCRIPTION
**Pull request by jscyo**
## Related Issues
- #193 

## Changes
- Working on:
   - SignInUpAPITest2_7 tests

## Migration:
  - none

### CODE REVIEW

File: src/test/java/io/supertokens/test/SignInUpAPITest2_7.java [hunk 1]
```diff
@@ -109,6 +109,85 @@ public void testEmailNormalisation() throws Exception {
...
+            requestBody.add("email", emailObject);
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/signinup", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_7ForTests(), ThirdParty.RECIPE_ID);
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                System.out.println(e.getMessage());
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/SignInUpAPITest2_7.java]: Remove ` System.out.println`. Make sure it's not there ANYWHERE in your code.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java [hunk 2]
```diff
@@ -45,4 +55,71 @@ public static void afterTesting() {
...
+            QueryParams.put("limit", "randomValue");
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendGETRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/users", QueryParams, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_7ForTests(), ThirdParty.RECIPE_ID);
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                System.out.println(e.getMessage());
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java:96]: Again ` System.out.println`...

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

File: src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java [hunk 3]
```diff
@@ -120,4 +120,113 @@ public void testBadInput() throws Exception {
...
+    @Test
+    public void testGoodInput() throws Exception {
+        String[] args = {"../"};
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        ThirdParty.signInUp(process.getProcess(), "thirdPartyId",
+                "thirdPartyUserId", "test@example.com", false);
+        Thread.sleep(100);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java:138]: Write a comment for why you have thread.sleep(100)

File: src/test/java/io/supertokens/test/ThirdPartyTest.java [hunk 4]
```diff
@@ -454,14 +454,19 @@ public void testGetUsers() throws Exception {
 
         ThirdParty.signInUp(process.getProcess(), "thirdPartyId",
                 "thirdPartyUserId", "test@example.com", false);
+        Thread.sleep(100);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/ThirdPartyTest.java:457]: Write a comment as to why you are pausing

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/test/java/io/supertokens/test/TelemetryAPITest2_7.java [hunk 5]
```diff
@@ -39,4 +48,97 @@ public void beforeEach() {
...
+
+        if (!Version.getVersion(process.getProcess()).getPluginName().equals("sqlite")) {
+            return;
+        }
+
+        JsonObject response = io.supertokens.test.httpRequest.HttpRequest
+                .sendGETRequest(process.getProcess(), "",
+                        "http://localhost:3567/telemetry", new HashMap<>(), 1000,
+                        1000,
+                        null, Utils.getCdiVersion2_7ForTests(), ThirdParty.RECIPE_ID);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/TelemetryAPITest2_7.java:66]: Recipe ID should be "" for all tests in this file

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/test/java/io/supertokens/test/TelemetryAPITest2_7.java (hunk 5 above)
Comment (jscyo) [reply src/test/java/io/supertokens/test/TelemetryAPITest2_7.java:66]: Alright making the change

File: src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java (hunk 3 above)
Comment (jscyo) [reply src/test/java/io/supertokens/test/ThirdPartyUsersAPITest2_7.java:139]: added comment to explain the addition of Thread.sleep

File: src/test/java/io/supertokens/test/ThirdPartyTest.java (hunk 4 above)
Comment (jscyo) [reply src/test/java/io/supertokens/test/ThirdPartyTest.java:458]: added comment

**Review by rishabhpoddar — APPROVED** [review]

File: src/test/java/io/supertokens/test/WebserverTest.java [hunk 6]
```diff
@@ -672,6 +758,80 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
         assertNotNull(process.checkOrWaitForEvent(PROCESS_STATE.STOPPED));
     }
 
+    public static RecipeRouter getRecipeRouter(TestingProcess process, String recipe_1, String recipe_2)
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/WebserverTest.java:761]: make this private

File: src/test/java/io/supertokens/test/WebserverTest.java [hunk 7]
```diff
@@ -71,6 +73,90 @@ public void beforeEach() {
...
+                            1000,
+                            null, Utils.getCdiVersion2_7ForTests(), recipe_2);
+            assertEquals("put request from Recipe2", response);
+
+            response = io.supertokens.test.httpRequest.HttpRequest
+                    .sendJsonDELETERequest(process.getProcess(), "",
+                            "http://localhost:3567/testRecipe", new JsonObject(), 1000,
+                            1000,
+                            null, Utils.getCdiVersion2_7ForTests(), recipe_2);
+            assertEquals("delete request from Recipe2", response);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/WebserverTest.java:153]: Add for when there is no rid given and `recipe_1` should be used

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/test/java/io/supertokens/test/WebserverTest.java (hunk 6 above)
Comment (jscyo) [reply src/test/java/io/supertokens/test/WebserverTest.java:797]: alright

File: src/test/java/io/supertokens/test/WebserverTest.java (hunk 7 above)
Comment (jscyo) [reply src/test/java/io/supertokens/test/WebserverTest.java:153]: alright

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**